### PR TITLE
upload four solutions for array problems

### PR DESCRIPTION
**Pull request by jdjin3000**
I solved below problems:

### Solved problem list 🔥
- Two Sum
https://leetcode.com/problems/two-sum/

- Trapping Rain Water
https://leetcode.com/problems/trapping-rain-water/

- 3Sum
https://leetcode.com/problems/3sum/

- Array Partition
https://leetcode.com/problems/array-partition/

### CODE REVIEW

File: array/3sum_DJ.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+                if nums[left] + nums[right] < -n:
+                    left +=1
+                elif nums[left] + nums[right] > -n:
+                    right -= 1
+                else: # if n + nums[left] + nums[right] == 0
+                    answer.add((n, nums[left], nums[right]))
+
+                    left += 1 # right -=1 이어도 상관 없음. 단지 다음 계산을 위한 방법.
+        
+        return sorted(map(list, list(answer)))
```
Comment (dobbytk) [inline array/3sum_DJ.py]: 문제에 'Notice that the order of the output and the order of the triplets does not matter.'라고 되어 있으니 return 할 때 sorted는 제외해도 되지 않을까 싶네요.

Comment (jdjin3000) [reply array/3sum_DJ.py]: 앗 그런 조건이... 수정하겠습니다 ㅎㅎ

File: array/array-partition_DJ.py [hunk 2]
```diff
@@ -0,0 +1,3 @@
+class Solution:
+    def arrayPairSum(self, nums: List[int]) -> int:
+        return sum(sorted(nums)[::2])
```
Comment (dobbytk) [inline array/array-partition_DJ.py:3]: 배열의 파티션을 슬라이싱을 이용해서 나눌 수도 있군요. 하나 배워갑니다.

**Review by dobbytk — APPROVED** [review]
Well done!

**Pull request merged**